src/Command/DrupalUpdate/UpdateScriptUtility.php

```diff
@@ -0,0 +1,317 @@
+   * @param $package
+   * @param $file_url
+   * @param $save_to
+   */
+  function download_remote_file($package, $file_url, $save_to) {
+    if($package == 'drupal'){
+      $this->download_remote_file_drupal_core($package, $file_url, $save_to);
+      return;
+    }
+    $content = file_get_contents($file_url);
+    file_put_contents($save_to . '/' . $package . '.tar.gz', $content);
+    try {
+      $phar = new PharData($save_to . '/' . $package . '.tar.gz');
+      $this->rrmdir($save_to . '/' . $package);
+      $phar->extractTo($save_to, NULL, TRUE); // extract all files
+    } catch (\Exception $e) {
+      // handle errors
+    }
+  }
+
+  /**
+   * Download and extract tar files in drupal core temp directory path.
+   * @param $package
+   * @param $file_url
+   * @param $save_to
+   */
+  function download_remote_file_drupal_core($package, $file_url, $save_to) {
+    $content = file_get_contents($file_url);
+    $folder_name = str_replace('.tar.gz', '', basename($file_url));
+    file_put_contents($save_to . '/' . $package . '.tar.gz', $content);
+    try {
+      $phar = new PharData($save_to . '/' . $package . '.tar.gz');
+      $this->rrmdir($save_to . '/' . $package);
+      $phar->extractTo($save_to, NULL, TRUE); // extract all files
+      rename($save_to . '/' . $folder_name, $save_to . '/drupal');
+    } catch (\Exception $e) {
+      // handle errors
+    }
+    if($package == 'drupal'){
+      // Replace the folder to inside docroot folder.
+      $this->coreUpdate($save_to . '/drupal');
+      $this->rrmdir($save_to);
+    }
+  }
+
+  /**
+   * Remove directory and sub directory.
+   * @param $dir
+   */
+  function rrmdir($dir) {
+    if (is_dir($dir)) {
+      $objects = scandir($dir);
+      foreach ($objects as $object) {
+        if ($object != "." && $object != "..") {
+          if (filetype($dir . "/" . $object) == "dir")
+                        $this->rrmdir($dir . "/" . $object);
+          else unlink   ($dir . "/" . $object);
+        }
+      }
+      reset($objects);
+      rmdir($dir);
+    }
+  }
+
+  /**
+   * After extraction copy code temp to main folder replace.
+   * Core modules, themes, profile.
+   * @param $actual_di_path
+   * @param $replace_dir_path
+   */
+  function coreUpdate($actual_di_path) {
+    $dir    = $actual_di_path;
+    $replace_dir_path = str_replace('/temp_drupal_core/drupal', '', $actual_di_path);
+    $files1 = array_diff(scandir($dir), ['.', '..']);
+
+    foreach($files1 as $r_dir => $c_dir){
+      $tm_path = $dir . "/" . $c_dir;
+      if(!in_array($c_dir, $this->ignore_updates_files)){
+        if(is_dir($tm_path)){
+          $this->custom_copy($dir . '/' . $c_dir, $replace_dir_path . '/' . $c_dir);
+          continue;
+        }
+        rename($dir . '/' . $c_dir, $replace_dir_path . '/' . $c_dir);
+      }
+    }
+  }
+
+  /**
+   * Copy folders and subfolder from temp to main folder.
+   * @param $src
+   * @param $dst
+   */
+  function custom_copy($src, $dst) {
```

Comment:
Please use Symfony for this rather than a custom solution. See https://symfony.com/doc/current/components/filesystem.html#mirror